### PR TITLE
[KYUUBI #5795][K8S] Support to cleanup the spark driver pod periodically

### LINKED ISSUE #5795: [TASK][EASY] Add a scheduled thread to trigger k8s spark driver pod clean up periodically
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

K8S spark driver pod clean up use guava cache as cleanup trigger with time-based eviction, but the eviction will not be triggered until any get/put operation happened. 

In our use case, wu use `Elastic Container Instance`, which are billed on a pay-as-you-go basis, to run spark on k8s. So the driver pod with `Finished` state is supports to be delete immediately.

### How should we improve?

Add a scheduled thread to evict the cache periodically.

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by liaoyt**
# :mag: Description
## Issue References 🔗

This pull request fixes #5795

## Describe Your Solution 🔧

Create a single daemon thread to traverse cache map periodically, which will evict expired cache and trigger a pod clean up operation.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [ ] I have performed a self-review
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [x] Pull request title is okay.
- [x] No license issues.
- [x] Milestone correctly set?
- [ ] Test coverage is ok
- [x] Assignees are selected.
- [x] Minimum number of approvals
- [x] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -1241,6 +1241,15 @@ object KyuubiConf {
       .stringConf
       .createWithDefault(KubernetesCleanupDriverPodStrategy.NONE.toString)
 
+  val KUBERNETES_SPARK_CLEANUP_TERMINATED_DRIVE_POD_IMMEDIATELY: ConfigEntry[Boolean] =
+    buildConf("kyuubi.kubernetes.spark.cleanupTerminatedDriverPodImmediately")
```
[the remark this answers is not in the document]
Comment (liaoyt) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1245]: SGTM

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Merging to master/1.8

**Comment by sudohainguyen** [comment]
I'm using v1.9 but this change seems not working 🤔 
is there any logs to check if the daemon is still active?

**Comment by liaoyt** [comment]
> I'm using v1.9 but this change seems not working 🤔 is there any logs to check if the daemon is still active? @liaoyt

I sorry currently there is not any logs to check if the deamon is still active.
but you can try check if there is a thread named 'pod-cleanup-trigger-thread'

**Comment by sudohainguyen** [comment]
just checked, there's no thread named "pod-cleanup-trigger-thread"

**Comment by sudohainguyen** [comment]
but still doesn't delete pods

**Comment by A-little-bit-of-data** [comment]
> pod-cleanup-trigger-thread

I use the v1.9.1 version and configure the following two parameters kyuubi.kubernetes.spark.cleanupTerminatedDriverPod.kind=ALL
kyuubi.kubernetes.terminatedApplicationRetainPeriod=PT1M
After the executor is deleted after the timeout, the driver pod is in the completed state, but it is still retained and needs to be manually deleted. I also checked the pod log, there's no thread named "pod-cleanup-trigger-thread"  @liaoyt

**Pull request closed without merging**